src/main/java/de/tum/in/www1/artemis/service/TutorParticipationService.java

```diff
@@ -0,0 +1,129 @@
+import de.tum.in.www1.artemis.repository.TutorParticipationRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Service Implementation for managing TutorParticipation.
+ */
+@Service
+@Transactional
+public class TutorParticipationService {
+
+    private final Logger log = LoggerFactory.getLogger(TutorParticipationService.class);
+    private final ExampleSubmissionRepository exampleSubmissionRepository;
+
+    @Value("${server.url}")
+    private String ARTEMIS_BASE_URL;
+
+    private final TutorParticipationRepository tutorParticipationRepository;
+
+    public TutorParticipationService(TutorParticipationRepository tutorParticipationRepository,
+                                     ExampleSubmissionRepository exampleSubmissionRepository) {
+        this.tutorParticipationRepository = tutorParticipationRepository;
+        this.exampleSubmissionRepository = exampleSubmissionRepository;
+    }
+
+    /**
+     * Save a tutorParticipations.
+     *
+     * @param tutorParticipation the entity to save
+     * @return the persisted entity
+     */
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public TutorParticipation save(TutorParticipation tutorParticipation) {
+        log.debug("Request to save TutorParticipation : {}", tutorParticipation);
+        return tutorParticipationRepository.saveAndFlush(tutorParticipation);
+    }
+
+    /**
+     * Get all the tutorParticipations.
+     *
+     * @return the list of entities
+     */
+    @Transactional(readOnly = true)
+    public List<TutorParticipation> findAll() {
+        log.debug("Request to get all TutorParticipations");
+        return tutorParticipationRepository.findAll();
+    }
+
+    /**
+     * Get all the tutorParticipations.
+     *
+     * @param pageable the pagination information
+     * @return the list of entities
+     */
+    @Transactional(readOnly = true)
+    public Page<TutorParticipation> findAll(Pageable pageable) {
+        log.debug("Request to get all TutorParticipations");
+        return tutorParticipationRepository.findAll(pageable);
+    }
+
+    /**
+     * Get one tutorParticipations by id.
+     *
+     * @param id the id of the entity
+     * @return the entity
+     */
+    @Transactional(readOnly = true)
+    public TutorParticipation findOne(Long id) {
+        log.debug("Request to get TutorParticipation : {}", id);
+
+        Optional<TutorParticipation> tutorParticipation = tutorParticipationRepository.findById(id);
+
+        return tutorParticipation.orElse(null);
+    }
+
+    @Transactional
```

Comment:
Why is it not readonly? This is counter intuitive for a find method. Please add JavaDoc comments for all public methods.

Reply:
It actually should be readonly, thanks for noticing!